drivers/shared/eventer/eventer.go

```diff
@@ -81,6 +81,15 @@ func (e *Eventer) iterateConsumers(event *drivers.TaskEvent) {
 	e.consumersLock.Lock()
 	filtered := e.consumers[:0]
 	for _, consumer := range e.consumers {
+
+		// prioritize checking if context is cancelled prior
+		// to attempting to forwarding events
+		// golang select evaluations aren't predictable
+		if consumer.ctx.Err() != nil {
+			close(consumer.ch)
+			continue
+		}
```

Comment:
Won't this potentially cause you to close `consumer.ch` twice?

Reply:
What triggering condition are you thinking about?  If context already expired at beginning of loop, it will close channel and skip, if it's not, it will behave just like before.

Reply:
I don't know how I didn't see the `continue` :astonished: during the first pass